sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/README.md

```diff
@@ -1,13 +1,179 @@
+    instrumentation_key = os.environ["AZURE_MONITOR_INSTRUMENTATION_KEY"]
+)
+```
+
 ## Key concepts
 
+Some of the key concepts for the Azure monitor exporter include:
+
+* [Opentelemetry][opentelemtry_spec]: Opentelemetry is a tool to instrument, generate, collect, and export telemetry data (metrics, logs, and traces) for analysis in order to understand your software's performance and behavior.
+
+* [Trace][trace_concept]: Trace can be thought of as a directed acyclic graph (DAG) of Spans, where the edges between Spans are defined as parent/child relationship.
+
+* [AzureMonitorSpanExporter][client_reference]: This is the object a user should first initialize to connect the namespace to send trace data.
+
+* [Exporter Options][exporter_options]: Options to configure Azure exporters. Includes connection_string, instrumentation_key, proxies, timeout etc.
+
+For more information about these resources, see [What is Azure Monitor?][product_docs].
+
 ## Examples
 
+The following sections provide several code snippets covering some of the most common tasks, including:
+
+* [Export hello world trace data](#export-hello-world-trace)
+* [Batch Export Spans](#batch-export-span-processor)
+
+
+### Export Hello World Trace
+
+```Python
+import os
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
+from microsoft.opentelemetry.exporter.azuremonitor import AzureMonitorSpanExporter
+
+# Callback function to add os_type: linux to span properties
+def callback_function(envelope):
+    envelope.data.base_data.properties["os_type"] = "linux"
+    return True
+
+exporter = AzureMonitorSpanExporter(
+    connection_string = os.environ["AZURE_MONITOR_CONNECTION_STRING"]
+)
+exporter.add_telemetry_processor(callback_function)
+
+trace.set_tracer_provider(TracerProvider())
+tracer = trace.get_tracer(__name__)
+span_processor = BatchExportSpanProcessor(exporter)
+trace.get_tracer_provider().add_span_processor(span_processor)
+
+with tracer.start_as_current_span("hello"):
+    print("Hello, World!")
+```
+
+### Batch Export Span Processor
+
+```Python
+import os
+import requests
+from opentelemetry import trace
+from opentelemetry.instrumentation.requests import RequestsInstrumentor
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
+
+from microsoft.opentelemetry.exporter.azuremonitor import AzureMonitorSpanExporter
+
+trace.set_tracer_provider(TracerProvider())
+tracer = trace.get_tracer(__name__)
+RequestsInstrumentor().instrument()
+span_processor = BatchExportSpanProcessor(
+    AzureMonitorSpanExporter(
+        connection_string = os.environ["AZURE_MONITOR_CONNECTION_STRING"]
+    )
+)
+trace.get_tracer_provider().add_span_processor(span_processor)
+```
+
 ## Troubleshooting
 
+### Logging
+
+- Enable by setting `logging_enable=True` when creating the client.
```

Comment:
This could confuse users here, need to add details about logging of the actual library, sounds like a magic flag that will enable sending logging telemetry

Reply:
i believe this can actually be removed for now